docs/dev/vagrant.rst

```diff
@@ -141,48 +141,160 @@ Problems
+
+		git clone https://github.com/learning-unlimited/ESP-Website.git
+		cd ESP-Website/
+		git checkout migration-to-python3
+		esp/update_deps.sh
+		cd ..
+		rm -rf ESP-Website/
+		logout
+
+7. 
+
+	Export the box you have to a .box file by running ``vagrant package --output ubuntu-*.box`` from your desktop terminal in ``devsite``, once again replacing the star with the correct version.
+
+8. 
+
+	Upload the .box file to the LU AWS S3 bucket.
+	If you don't have access, ask someone on the LU Web Team.
+	When you upload it, choose "Choose from predefined ACLs" and "Grant public-read access" under "Permissions" at the bottom.
+
+9. 
+
+	Restore the vagrantfile by running ``git restore Vagrantfile``, and update it so that ``config.vm.box`` matches the box name (probably ``'ubuntu-*'``) and ``config.vm.box_url`` points to the new VM's URL (which you can copy from AWS).
+	Make sure to commit the changes in ``Vagrantfile`` to GitHub!
+
+10. 
+
+	Test that the new setup works.
+	From the ``devsite/`` directory, run ``rm -rf ~/.vagrant.d/ && rm -rf .vagrant && vagrant destroy -f && vagrant up && fab setup && fab emptydb``.
+	Again, this will remove other vagrant machines, so if you have others and know what you're doing, delete only the current one.
 
 Upgrading your personal dev VM
 ------------------------------
 
 If the base VM has been changed (see above), you will want to upgrade your development server. However, upgrading Ubuntu within a virtual machine can cause problems with your database. Therefore, you'll need to export your database, create a new virtual machine, then import your database:
 
-1. Make a copy of `esp/esp/local_settings.py` somewhere with a different name (e.g. on your desktop as "old_local_settings.py"). The original `local_settings.py` file will get overwitten by the end of this process and you will want to restore some of the settings from your previous VM setup.
+1. Make a copy of `esp/esp/local_settings.py` somewhere with a different name (e.g., on your desktop as "old_local_settings.py"). The `local_settings.py` file will get overwritten by the end of this process and you will want to restore the settings from your previous VM setup.
 
 2. From within the "devsite" folder, run ``fab dumpdb``. This action will save your database as a dump file in the `devsite` folder called `devsite_django.sql`. You can also specify a filename if you would like with ``fab dumpdb:filename``.
 
-3. Run ``vagrant destroy`` (note: this destroys your virtual machine. Only do it once you are sure your database has been backed up and you are ready to continue).
+3. Run ``vagrant destroy`` (note: this destroys your virtual machine. Only do it once you are sure your database has been backed up and you are ready to continue). 
 
 4. Now follow the `VM installation instructions above <#installation>`_, starting at ``vagrant up``.
+If you run into trouble, clear your SSH keys in ``~/.vagrant.d/`` and ``devsite/.vagrant``.
```

Comment:
Is this only if you have trouble with the vagrant up step, or is it also if you have trouble with the fab setup step?

vagrant up was fine, but I'm having trouble with fab setup. 

I got this error:

![image](https://github.com/learning-unlimited/ESP-Website/assets/25392075/1738f1b5-82b6-4b1d-a332-0da0b1b905ee)


so then i tried to drop the tablespace encryption but it didn't work:

![image](https://github.com/learning-unlimited/ESP-Website/assets/25392075/7583faf6-211b-40b2-97b1-4da8eae97e7b)

Reply:
Clearing keys only help with Vagrant. To drop the tablespace, you'd want to do `sudo -u postgres psql -c ...`. I'm surprised the tablespace already exists the first time you ran fab setup (unless you did it once before and it died mid-stream?) We can troubleshoot together

Reply:
Try again? I updated the  VM